### PR TITLE
Add CreateRootSignature

### PR DESCRIPTION
**Pull request by john-h-k**
Discussed in discord
Adds a tiny little wrapper method normally part of `d3dx12.h`
Native reference:
```cpp
inline HRESULT CreateRootSignature(
        _In_ ID3D12Device* device,
        _In_ const D3D12_ROOT_SIGNATURE_DESC* rootSignatureDesc,
        _Out_ ID3D12RootSignature** rootSignature) noexcept
    {
        Microsoft::WRL::ComPtr<ID3DBlob> pSignature;
        Microsoft::WRL::ComPtr<ID3DBlob> pError;
        HRESULT hr = D3D12SerializeRootSignature(rootSignatureDesc, D3D_ROOT_SIGNATURE_VERSION_1, pSignature.GetAddressOf(), pError.GetAddressOf());
        if (SUCCEEDED(hr))
        {
            hr = device->CreateRootSignature(0, pSignature->GetBufferPointer(), pSignature->GetBufferSize(),
                IID_GRAPHICS_PPV_ARGS(rootSignature)
                );
        }
        return hr;
    }
```

### CODE REVIEW

File: sources/Interop/D3D12/um/d3dx12/D3D12.Manual.cs [hunk 1]
```diff
@@ -23,6 +23,28 @@ public static unsafe partial class D3D12
             Count = 1,
             Quality = 0,
         };
+        
+        [return: NativeTypeName("HRESULT")]
+        public static int CreateRootSignature([NativeTypeName("ID3D12Device*")] ID3D12Device* device, [NativeTypeName("const D3D12_ROOT_SIGNATURE_DESC*")] D3D12_ROOT_SIGNATURE_DESC* rootSignatureDesc, [NativeTypeName("ID3D12RootSignature**")] ID3D12RootSignature** rootSignature)
```
Comment (tannergooding) [inline sources/Interop/D3D12/um/d3dx12/D3D12.Manual.cs:28]: Clang normalizes to have a space between the type and the `*`
```suggestion
        public static int CreateRootSignature([NativeTypeName("ID3D12Device *")] ID3D12Device* device, [NativeTypeName("const D3D12_ROOT_SIGNATURE_DESC *")] D3D12_ROOT_SIGNATURE_DESC* rootSignatureDesc, [NativeTypeName("ID3D12RootSignature **")] ID3D12RootSignature** rootSignature)
```

File: sources/Interop/D3D12/um/d3dx12/D3D12.Manual.cs [hunk 2]
```diff
@@ -23,6 +23,43 @@ public static unsafe partial class D3D12
...
+        {
+            ID3DBlob* pSignature = default;
+            ID3DBlob* pError = default;
+            
+            try
```
Comment (tannergooding) [inline sources/Interop/D3D12/um/d3dx12/D3D12.Manual.cs:33]: Why is the `try` needed? The native code doesn't have one and we aren't throwing an exception here, in fact we are explicitly not throwing and returning the `HRESULT`

Comment (john-h-k) [reply sources/Interop/D3D12/um/d3dx12/D3D12.Manual.cs:33]: I'll remove, I just saw the ComPtr and blindly added a finally as equivalent to the `Microsoft::WRL::ComPtr` dtor

**Comment by tannergooding** [comment]
As per our Discord conversation, this isn't actually part of D3DX12 and is instead part of DX12TK; so the PR should be adjusted to account for that.

**Review by tannergooding — APPROVED** [review]

**Review by tannergooding — APPROVED** [review]

**Pull request merged**